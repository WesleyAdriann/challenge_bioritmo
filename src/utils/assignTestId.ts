type componentType = 'div' | 'text' | 'input' | 'button' | 'img'

export const assignTestId = (
  componentType: componentType,
  componentId: string
)  => ({
    'data-testid': `${componentType}_${componentId}`,
  })


export const mockTestID = (
    componentType: componentType,
    componentId: string
  ) => assignTestId(componentType, componentId)
