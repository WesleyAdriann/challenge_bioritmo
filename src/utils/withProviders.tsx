
import React from 'react'

import { ThemeProvider } from 'styled-components'

import { theme } from '../style'

export const StorybookProviders: React.FC = ({
  children,
}) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
)
