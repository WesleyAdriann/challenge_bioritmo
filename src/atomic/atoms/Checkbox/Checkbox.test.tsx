import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'

import { mockTestID, StorybookProviders as wrapper } from '../../../utils'

import { Checkbox, ICheckbox } from './index'

describe('Atoms/Checkbox', () => {
  const props: ICheckbox = {
    testID: 'Checkbox',
    active: true,
    onClick: jest.fn()
  }

  const renderComponent = () => render(
    <Checkbox {...props} />, { wrapper }
  )

  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  const el_container = mockTestID('div', props.testID!)

  it(`should render component #${el_container}`, async () => {
    const sut = await component.findByTestId(el_container)
    expect(sut).toBeTruthy()
  })

  it('should render check icon in Checkbox with active true', async () => {
    component.rerender(<Checkbox {...props} />)
    const sut = await component.findByTestId(el_container)
    expect(sut.children.length).toBeTruthy()
  })

  it('should not render icon in Checkbox with active false', async () => {
    component.rerender(<Checkbox {...props} active={false} />)
    const sut = await component.findByTestId(el_container)
    expect(sut.children.length).toBeFalsy()
  })

  it('should call onClick prop on click in checkbox', async () => {
    const sut = await component.findByTestId(el_container)
    fireEvent.click(sut)
    expect(props.onClick).toBeCalled()
  })

  it('should render snapshot', () => {
    expect(component.container.firstChild).toMatchSnapshot()
  })
})
