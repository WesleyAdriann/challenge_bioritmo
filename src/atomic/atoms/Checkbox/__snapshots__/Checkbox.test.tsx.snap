// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Checkbox should render snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFB612;
  border: 3px solid #808080;
  border-radius: 2px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1px;
  width: 24px;
}

.c0:active {
  -webkit-filter: brightness(105%);
  filter: brightness(105%);
}

<div
  class="c0"
  data-testid="div_Text"
  size="24"
>
  <svg
    class=""
    color="#000"
    height="24px"
    size="24"
    viewBox="0 0 24 24"
    width="24px"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
      fill="#000"
    />
  </svg>
</div>
`;
